core/commands/block.go

```diff
@@ -113,32 +113,41 @@ It outputs to stdout, and <key> is a base58 encoded multihash.
 }
 
 const (
-	blockFormatOptionName = "format"
-	mhtypeOptionName      = "mhtype"
-	mhlenOptionName       = "mhlen"
+	blockFormatOptionName   = "format"
+	blockCidCodecOptionName = "cid-codec"
+	mhtypeOptionName        = "mhtype"
+	mhlenOptionName         = "mhlen"
 )
 
 var blockPutCmd = &cmds.Command{
 	Helptext: cmds.HelpText{
 		Tagline: "Store input as an IPFS block.",
 		ShortDescription: `
 'ipfs block put' is a plumbing command for storing raw IPFS blocks.
-It reads from stdin, and outputs the block's CID to stdout.
+It reads data from stdin, and outputs the block's CID to stdout.
 
-Unless specified, this command returns dag-pb CIDv0 CIDs. Setting 'mhtype' to anything
-other than 'sha2-256' or format to anything other than 'v0' will result in CIDv1.
+Unless cid-codec is specified, this command returns raw (0x55) CIDv1 CIDs.
+
+Passing alternative --cid-codec does not modify imported data, nor run any
+validation. It is provided solely for convenience for users who create blocks
+in userland.
+
+NOTE:
+Do not use --format for any new code. It got superseded by --cid-codec and left
+only for backward compatibility when a legacy CIDv0 is required (--format=v0).
 `,
 	},
 
 	Arguments: []cmds.Argument{
 		cmds.FileArg("data", true, true, "The data to be stored as an IPFS block.").EnableStdin(),
 	},
 	Options: []cmds.Option{
-		cmds.StringOption(blockFormatOptionName, "f", "cid format for blocks to be created with."),
-		cmds.StringOption(mhtypeOptionName, "multihash hash function").WithDefault("sha2-256"),
-		cmds.IntOption(mhlenOptionName, "multihash hash length").WithDefault(-1),
-		cmds.BoolOption(pinOptionName, "pin added blocks recursively").WithDefault(false),
+		cmds.StringOption(blockCidCodecOptionName, "Multicodec to use in returned CID").WithDefault("raw"),
```

Comment:
@aschmahmann I did some tests around this and the good news is that I was wrong, there is a way to have unified handling of default opts across `ipfs block put` CLI and raw HTTP POST to `/api/v0/block/put`.

If we keep the `.WithDefault("raw")` here, then this implicit default is also applied when people use curl against `/api/v0/block/put` and forget to pass `cid-codec`. 

If we want to be extra strict and return error, I'd have to remove  `WithDefault` and add some glue code that detects `cidCodec == ""`, but that feels hacky  and annoying because it would make `cid-codec` mandatory on CLI too.

Reply:
Added to CHANGELOG.md in https://github.com/ipfs/go-ipfs/pull/8568/commits/2f5b8a4d9697e909da9f015fe23dac946d171d37